ghcide/test/exe/Config.hs

```diff
@@ -31,9 +34,15 @@ runWithDummyPlugin' = runSessionWithServerInTmpDirCont' def dummyPlugin
 runWithDummyPluginAndCap :: ClientCapabilities -> Session () -> IO ()
 runWithDummyPluginAndCap cap = runSessionWithServerAndCapsInTmpDir def dummyPlugin cap (mkIdeTestFs [])
 
+runWithDummyPluginAndCap' :: ClientCapabilities -> (FileSystem -> Session ()) -> IO ()
```

Comment:
FWIW, I'm not sure all these functions are necessarily worth it. How bad would it be to just have the `FileSystem -> Session ()` variants and then write `_ -> whatever` at the use sites?

Reply:
Emm, we are at somewhat funny situation, that the one without the extra FileSystem is the most used one. 
Sure, for not frequent used ones, it would be fine we write `_ -> whatever`.

Reply:
I guess I'm wondering how bad it would be. Otherwise we have a lot of very similar variant functions :thinking:

Reply:
Not a blocker, anyway

Reply:
Yep, I mean since we have a lot of test that does not need the `FileSystem `,
adding a tons of `_ -> whatever` is visually bad and and not convenient for us to do the migration.

That is why I want to either do the typeclass polymorphic or just duplicate the test runner function variant.